src/NServiceBus.Core/Pipeline/Outgoing/InspectOutgoingMessageBehavior.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    class InspectOutgoingMessageBehavior : Behavior<IOutgoingPhysicalMessageContext>
```

Comment:
@danielmarbach I tested adding this code to ```ImmediateDispatchTerminator``` and the issue is that it is being also called by AuditMessage. Which means that setting time_sent happens multiple times. So in the audit message it is time of sending AuditMessage not the original.

In a way I believe that having separate behavior for this seems like a bit overkill but the above solution would need to be suplemented with an if ;) I am open for discussion